engine/gamesys/src/gamesys/scripts/script_resource.cpp

```diff
@@ -673,7 +722,7 @@ static int CreateTexture(lua_State* L)
     void* resource = 0x0;
     dmResource::Result res = dmResource::CreateResource(g_ResourceModule.m_Factory, path_str, ddf_buffer.Begin(), ddf_buffer.Size(), &resource);
 
-    DestroyTextureImage(texture_image);
+    DestroyTextureImage(texture_image, buffer == 0);
```

Comment:
If the script passed in a buffer we shouldn't delete the data since the buffer manages that